core/sql/optimizer/NAColumn.cpp

```diff
@@ -749,6 +749,27 @@ Lng32 NAColumnArray::getOffset(Lng32 position) const
 
   return result;
 }
+
+Lng32 NAColumnArray::getMaxTrafHbaseColQualifier() const
+{
+  NAColumn *column;
+  char * colQualPtr;
+  Lng32 colQualLen;
+  Int64 colQVal;
+  Lng32 maxVal = 0;
+
+  for (CollIndex i = 0; i < entries(); i++)
+    {
+      column = (*this)[i];
+      colQualPtr = (char*)column->getHbaseColQual().data();
+      colQualLen = column->getHbaseColQual().length();
+      colQVal = str_atoi(colQualPtr, colQualLen);
+      if (colQVal > maxVal)
+	maxVal = colQVal ;
+    }
+  return colQVal;
```

Comment:
Thanks for catching this. I will fix the return value in a minute. The smallest colQualifier we add is 1 for a Traf table. We don't seem to add colQualifier with value 0. Returning a negative value will mean that it will have to be checked by the caller.